app/src/main/java/pt/ulisboa/tecnico/cmov/freelibrary/LibraryInfo.java

```diff
@@ -161,4 +234,76 @@ public void onFailure(Call<List<Book>> call, Throwable t) {
             }
         });
     }
+
+    private void fetchLibraries() {
+        apiService.getAllLibraries().enqueue(new Callback<List<Library>>() {
+            @Override
+            public void onResponse(Call<List<Library>> call, Response<List<Library>> response) {
+                if (response.isSuccessful()) {
+                    List<Library> libraries = response.body();
+                    for (Library library : libraries) {
+                        if (library.id == libraryId) {
+                            libraryLocation = new LatLng(library.latitude, library.longitude);
+                            mGoogleMap.addMarker(new MarkerOptions().position(libraryLocation).title(library.name));
+                            LatLngBounds.Builder builder = new LatLngBounds.Builder();
+                            builder.include(libraryLocation);
+                            bounds = builder.build();
+                            int padding = 100; // Adjust the padding as needed
+                            mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngBounds(bounds, padding));
```

Comment:
Padding of 15 would be probably better
mGoogleMap.animateCamera(CameraUpdateFactory.newLatLngZoom(latlng, 15));

Reply:
Done